docs/howtos/ckeditor.md

```diff
@@ -2,7 +2,127 @@
+
+## Changing the allowed HTML tags in rich text
+
+Some CKEditor configurations introduce new HTML tags that might be present in the markup. It looks great in the editor, but when you refresh the page they are gone. What happened?
+
+Apostrophe automatically filters all rich text through the [sanitize-html](https://npmjs.org/package/sanitize-html) module. This prevents XSS attacks and also prevents the page design from being wrecked by bad or just plain ugly markup pasted from desktop applications.
+
+However, sometimes the default configuration isn't working for you, for instance because you want to add `sub` and `sup` to the list of allowed tags.
+
+It's easy to change: just set the `sanitizeHtml` option in `lib/modules/apostrophe-rich-text-widgets/index.js` in your project. Just keep in mind that **if you configure one of the sanitizeHtml options at all, you must configure that option completely.** A common mistake is forgetting to allow `a` elements or the `http` protocol, resulting in very pretty text with no links allowed!
+
+Here are a few common configurations. For more, [see the `sanitize-html` documentation](https://npmjs.org/package/sanitize-html).
+
+### Add text styles using classes
+
+You will often want to add styles to the rich text editor widget that use classes for visual styling. For example, you may have these configurations in an area with a rich text widget for a basic paragraph and two special styles:
+
+```django
+  {{ apos.area(data.page, 'body', {
+    widgets: {
+      'apostrophe-rich-text': {
+        toolbar: [ 'Styles', 'Bold', 'Italic', 'Link', 'Unlink' ],
+        styles: [
+          { element: 'p', name: 'Paragraph' },
+          {
+            element: 'p',
+            name: 'Featured text',
+            attributes: { class: 'featured-text' }
+          },
+          {
+            element: 'h3',
+            name: 'Section title',
+            attributes: { class: 'section-heading' }
+          }
+        ]
+      }
+    }
+  }) }}
+```
+
+You can allow these classes in two ways. First, you could allow all classes on those elements using `allowedAttributes`:
+
+```javascript
+// lib/modules/apostrophe-rich-text-widgets/index.js
+
+module.exports = {
+  // The standard list copied from the module, plus sup and sub
+  sanitizeHtml: {
+    // allowedTags: [...],
+    allowedAttributes: {
+      'p': ['class'],
+      'h3': ['class'],
+      // Include the default setting as well, or else links will break
+      'a': [ 'href', 'name', 'target' ]
+    },
+    // ...,
+  }
+};
+```
+
+::: tip
+You can open this up to allow the `class` attribute on any element by replacing those individual tag name keys in `allowedAttributes` with an asterisk string (`'*': ['class']`).
+:::
+
+Alternatively, you could only allow specific classes. You may not want to allow people to paste in rich text from somewhere else that includes classes that don't work well in a certain context. In this approach, you would use `allowClasses`:
```

Comment:
Is it `allowClasses` or `allowedClasses`? I see it both ways here.

Reply:
good catch. It's `allowedClasses`.